### PR TITLE
Add support for ot-metrics

### PR DESCRIPTION
**Pull request by skef**
Some test infrastructure I'm fooling with would benefit from this.

Hoping you might have some advice on whether or how I should test the x/y_variation functions, or if I should just delete those ...

### CODE REVIEW

**Comment by anthrotype** [comment]
thanks!

> some advice on whether or how I should test the x/y_variation functions, or if I should just delete those ...

what's different/special about those as far as testing is concerned?

**Comment by skef** [comment]
> what's different/special about those as far as testing is concerned?

Looking at the source, they seem to scale the returned values based on `em_scalef_x` or `em_scalef_y`, which use scaling factors in the font. But I didn't dig too deeply on how to set those factors. @khaledhosny might have a better idea.

**Comment by skef** [comment]
After thinking about this I added a mundane test for each of the methods. Given that uharfbuzz is a very thin layer over HarfBuzz, that's probably fine. Even if these methods aren't useful now, perhaps future additions will make them so.

**Comment by khaledhosny** [comment]
I’m not familiar with this API, to be honest.

**Comment by anthrotype** [comment]
fwiw i'd be fine with merghing this as is

**Comment by skef** [comment]
If it passes review, including verifying that I'm calling each method I added in the tests in a somewhat substantive way, I think it will be fine.

File: src/uharfbuzz/_harfbuzz.pyx [hunk 1]
```diff
@@ -1624,6 +1624,73 @@ def ot_math_get_glyph_assembly(font: Font,
...
+    SUPERSCRIPT_EM_Y_SIZE = HB_OT_METRICS_TAG_SUPERSCRIPT_EM_Y_SIZE
+    SUPERSCRIPT_EM_X_OFFSET = HB_OT_METRICS_TAG_SUPERSCRIPT_EM_X_OFFSET
+    SUPERSCRIPT_EM_Y_OFFSET = HB_OT_METRICS_TAG_SUPERSCRIPT_EM_Y_OFFSET
+    STRIKEOUT_SIZE = HB_OT_METRICS_TAG_STRIKEOUT_SIZE
+    STRIKEOUT_OFFSET = HB_OT_METRICS_TAG_STRIKEOUT_OFFSET
+    UNDERLINE_SIZE = HB_OT_METRICS_TAG_UNDERLINE_SIZE
+    UNDERLINE_OFFSET = HB_OT_METRICS_TAG_UNDERLINE_OFFSET
+
+def ot_metrics_get_position(font: Font,
+                            tag: OTMathTag) -> int:
```
Comment (khaledhosny) [inline src/uharfbuzz/_harfbuzz.pyx:1659]:
```suggestion
                            tag: OTMetricsTag) -> int:
```

**Comment by skef** [comment]
Sorry, don't know where my head was on this one ...

**Review by khaledhosny — APPROVED** [review]

**Pull request merged**